python/lsst/ts/MTAlignment/mock/mock_t2sa.py

```diff
@@ -0,0 +1,926 @@
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <https://www.gnu.org/licenses/>.
+
+__all__ = ["MockT2SA"]
+
+import asyncio
+import logging
+import re
+import typing
+
+import numpy as np
+
+from lsst.ts import tcpip
+from lsst.ts import utils
+from .mock_t2sa_point_group import MockT2SAPointGroup
+from ..utils import CartesianCoordinate, BodyRotation
+
+# Good reply bodies. They should come after the initial "ACK-300 " in replies,
+# and thus are intended as arguments to `write_good_reply`.
+ALREADY_MEASURING_REPLY = "ACK000"
+OK_REPLY = "ACK300"  # Command was accepted
+
+
+class MockT2SA(tcpip.OneClientServer):
+    """Emulate a New River Kinematics T2SA application.
+
+    This emulator relies on a circular body representation of each individual
+    "point group" in the T2SA application to provide realistic measurements.
+
+    Responses to messages sent by the client are parsed using a regular
+    expression and used to call appropriate methods do handle the responses.
+    In few cases the mock relies on a simple canned response.
+
+    Parameters
+    ----------
+    host : `str` or `None`
+        IP address for this server; typically `LOCAL_HOST`
+    port : `int`
+        IP port for this server. If 0 then randomly pick an available port
+        (or ports, if listening on multiple sockets).
+        0 is strongly recommended for unit tests.
+    log : `logging.Logger`
+        Logger; if None create a new one.
+
+    Attributes
+    ----------
+    host : `str` or `None`
+        IP address for this server; typically `LOCAL_HOST`
+    port : `int`
+        IP port for this server. If this mock was constructed with port=0
+        then this will be the assigned port, once `start` is done.
+    """
+
+    def __init__(
+        self,
+        log: logging.Logger,
+        host: str = tcpip.LOCAL_HOST,
+        port: int = 0,
+    ) -> None:
+        self.measure_task = utils.make_done_future()
+        self.reply_loop_task = utils.make_done_future()
+
+        self.laser_warmup_task: asyncio.Task = utils.make_done_future()
+
+        self.canned_replies = {
+            "!SET_SIM:0": "ACK300",
+            "!SET_SIM:1": "ACK300",
+            "SET_RANDOMIZE_POINTS:0": "ACK300",
+            "SET_RANDOMIZE_POINTS:1": "ACK300",
+            "!RESET_T2SA": "ACK300",
+            "!NEW_STATION": "ACK300",
+        }
+
+        self.dispatchers: dict[str, tuple[typing.Any, re.Pattern]] = {
+            name: (method, re.compile(regex_str))
+            for name, method, regex_str in (
+                ("!2FACE_CHECK", self.execute_two_face_check, r"(?P<point_group>.*)"),
```

Comment:
I strongly suggest using a standard prefix for all command handlers, e.g. "do_" or "handle_". That makes the code a bit easier to understand, since one can tell at a glance whether a particular method is a command handler.